cni/network/invoker_azure.go

```diff
@@ -131,35 +139,38 @@ func (invoker *AzureIPAMInvoker) Delete(address *net.IPNet, nwCfg *cni.NetworkCo
 		nwCfg.IPAM.Subnet = invoker.nwInfo.Subnets[0].Prefix.String()
 	}
 
-	if address == nil {
+	if len(addresses) == 0 {
 		if err := invoker.plugin.DelegateDel(nwCfg.IPAM.Type, nwCfg); err != nil {
 			return invoker.plugin.Errorf("Attempted to release address with error:  %v", err)
 		}
-	} else if len(address.IP.To4()) == 4 {
-		nwCfg.IPAM.Address = address.IP.String()
-		log.Printf("Releasing ipv4 address :%s pool: %s",
-			nwCfg.IPAM.Address, nwCfg.IPAM.Subnet)
-		if err := invoker.plugin.DelegateDel(nwCfg.IPAM.Type, nwCfg); err != nil {
-			log.Printf("Failed to release ipv4 address: %v", err)
-			return invoker.plugin.Errorf("Failed to release ipv4 address: %v", err)
-		}
-	} else if len(address.IP.To16()) == 16 {
-		nwCfgIpv6 := *nwCfg
-		nwCfgIpv6.IPAM.Environment = common.OptEnvironmentIPv6NodeIpam
-		nwCfgIpv6.IPAM.Type = ipamV6
-		nwCfgIpv6.IPAM.Address = address.IP.String()
-		if len(invoker.nwInfo.Subnets) > 1 {
-			nwCfgIpv6.IPAM.Subnet = invoker.nwInfo.Subnets[1].Prefix.String()
-		}
+	}
+
+	for _, address := range addresses {
```

Comment:
Have you tested these changes in a non-overlay, swift multitenancy setup?
See this doc for more: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-networking/sdn-dbansal/azure-container-networking/azure-container-networking-docs/docs/developerguide/networkingaquarius/pipelines/standalonedev

Reply:
We just want to make sure they don't regress anything, here's what you do:
1. make a dev cluster like in the doc linked above ^
2. Delegate subnet
3. Run tests using YOUR CNI version, make sure they pass
Any questions, refer to the doc and linked recording in the doc